### PR TITLE
feat: connect wallet to web2 account on action

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-web#2159

Prompt web2 user to connect a wallet for on-chain action

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. https://deploy-preview-2219--regen-marketplace.netlify.app/

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-marketplace/src/pages/ProfileEdit/hooks/useConnectWalletToAccount.tsx [hunk 1]
```diff
@@ -76,5 +75,19 @@ export const useConnectWalletToAccount = ({
...
+
+  const connectWalletCallback = useCallback(async () => {
+    isConnectingWalletRef.current = true;
+    await connectWalletToAccount();
+    isConnectingWalletRef.current = false;
+  }, [connectWalletToAccount]);
+
+  useEffect(() => {
+    if (shouldConnectAccount && !isConnectingWalletRef.current) {
+      connectWalletCallback();
```
Comment (blushi) [inline web-marketplace/src/pages/ProfileEdit/hooks/useConnectWalletToAccount.tsx:90]: I'm getting the 'prove ownership' popup multiple times now

**Comment by erikalogie** [comment]
> > There seems to be an issue on the deploy preview with the google login. It might be best to test locally in the meantime.
> 
> I've updated the env var in Heroku so it redirects to the deploy preview for logging in with google.
> 
> @erikalogie you can test this on https://deploy-preview-2219--regen-marketplace.netlify.app/ Login with google and then try to perform on chain actions like buy, take from/put in basket...

I can't log in with google, I've tried all my different accounts and it isn't working.


https://github.com/regen-network/regen-web/assets/8460237/1351ddf4-946c-40a0-8067-569db9664797

**Comment by blushi** [comment]
> > > There seems to be an issue on the deploy preview with the google login. It might be best to test locally in the meantime.
> > 
> > 
> > I've updated the env var in Heroku so it redirects to the deploy preview for logging in with google.
> > @erikalogie you can test this on https://deploy-preview-2219--regen-marketplace.netlify.app/ Login with google and then try to perform on chain actions like buy, take from/put in basket...
> 
> I can't log in with google, I've tried all my different accounts and it isn't working.
> 
>  Screen.Recording.2023-11-22.at.8.41.55.AM.mov

Weird, still investigating. In the meantime, you should be able to log in with email/passcode now.

**Comment by erikalogie** [comment]
> > > > There seems to be an issue on the deploy preview with the google login. It might be best to test locally in the meantime.
> > > 
> > > 
> > > I've updated the env var in Heroku so it redirects to the deploy preview for logging in with google.
> > > @erikalogie you can test this on https://deploy-preview-2219--regen-marketplace.netlify.app/ Login with google and then try to perform on chain actions like buy, take from/put in basket...
> > 
> > 
> > I can't log in with google, I've tried all my different accounts and it isn't working.
> > Screen.Recording.2023-11-22.at.8.41.55.AM.mov
> 
> Weird, still investigating. In the meantime, you should be able to log in with email/passcode now.

Great yes it is working properly with email/passcode log in

**Comment by erikalogie** [comment]
I did notice this small typo (was my bad, it was in the comps) https://monosnap.com/file/Hi4bf1EQnLTsVF03Xnw5qg6UqurgG6

**Comment by flagrede** [comment]
@blushi I think you had several 'prove ownership' popups because you had an error (like address already used) but the error was not handled properly because of a change made in `postData`. I fixed that here [d4d83cb](https://github.com/regen-network/regen-web/pull/2219/commits/d4d83cb646bf77641661768a95f5c6e633407a61), should be better.

**Comment by blushi** [comment]
> @blushi Also I noted that the google login on staging does not redirect to the good origin preview URL. The login works but you would have to come back manually to the good preview url.

yeah this is because the app redirect URL is set to the deploy preview URL for https://github.com/regen-network/regen-web/pull/2221 right now

**Comment by flagrede** [comment]
> **marie left a comment:**
> 
> If I try to connect a wallet address (assuming I haven't connected yet, auto_connect_wallet is not set), then I don't even get the prove ownership popup, only the wallet address is set from Keplr.
> 
> recording
> 
> Browser metadata
> [Open Deploy Preview](https://deploy-preview-2219--regen-marketplace.netlify.app/profile/edit/settings) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=656469a7099ee70008c65ccb&commentID=6564860125e19436f4f47e36&resolution=resolved)

This is not the same story. This story is about connecting an account if the user tries to perform an on-chain action (like buying credits). That said, we might have a regression on this part in the user profile settings.

**Review by blushi — APPROVED** [review]
tACK

**Pull request merged**